src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int COEFFICIENT = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY = 16;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> currentNode = new Node<>(key, value, null);
+        Node<K, V> node = table[index];
+        if (node == null) {
+            table[index] = currentNode;
+        }
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = currentNode;
+                break;
+            }
+            node = node.next;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = table[getIndex(key)];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
     }
+
+    private void resize() {
+        Node<K, V>[] nodesArray = table;
+        size = 0;
+        table = new Node[nodesArray.length * COEFFICIENT];
+        for (Node<K, V> node : nodesArray) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
```

Comment:
I think it is better to move this logic to separate method.

Reply:
I understand, but if I move this logic to another method, I will need to change logic of another methods in my program.